applications/FluidDynamicsApplication/custom_elements/symbolic_stokes.cpp

```diff
@@ -0,0 +1,4943 @@
+        this->CalculateGeometryData(gauss_weights, shape_functions, shape_derivatives);
+        const unsigned int number_of_gauss_points = gauss_weights.size();
+
+        double sumOfGaussWeights = 0.0;
+
+        for (unsigned int g = 0; g < number_of_gauss_points; g++){
+
+            UpdateIntegrationPointData(dataLocal, g, gauss_weights[g], row(shape_functions, g), shape_derivatives[g]);
+
+            const Vector gauss_point_contribution = dataLocal.ShearStress;
+
+            noalias( rOutput ) += gauss_point_contribution * gauss_weights[g];
+            sumOfGaussWeights += gauss_weights[g];
+        }
+
+        for (unsigned int i = 0; i < StrainSize; i++){
+            rOutput[i] = ( 1.0 / sumOfGaussWeights ) * rOutput[i];
+        }
+
+    } else {
+
+        Element::Calculate(rVariable, rOutput, rCurrentProcessInfo);
+
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Protected operations
+
+template <class TElementData>
+void SymbolicStokes<TElementData>::AddTimeIntegratedSystem(
+    TElementData &rData,
+    MatrixType &rLHS,
+    VectorType &rRHS)
+{
+    this->ComputeGaussPointLHSContribution(rData, rLHS);
+    this->ComputeGaussPointRHSContribution(rData, rRHS);
+}
+
+template <class TElementData>
+void SymbolicStokes<TElementData>::AddTimeIntegratedLHS(
+    TElementData &rData,
+    MatrixType &rLHS)
+{
+    this->ComputeGaussPointLHSContribution(rData, rLHS);
+}
+
+template <class TElementData>
+void SymbolicStokes<TElementData>::AddTimeIntegratedRHS(
+    TElementData &rData,
+    VectorType &rRHS)
+{
+    this->ComputeGaussPointRHSContribution(rData, rRHS);
+}
+
+template <class TElementData>
+void SymbolicStokes<TElementData>::UpdateIntegrationPointData(
+    TElementData& rData,
+    unsigned int IntegrationPointIndex,
+    double Weight,
+    const typename TElementData::MatrixRowType& rN,
+    const typename TElementData::ShapeDerivativesType& rDN_DX) const
+{
+    rData.UpdateGeometryValues(IntegrationPointIndex, Weight, rN, rDN_DX);
+    this->CalculateMaterialResponse(rData);
+}
+
+template <>
+void SymbolicStokes<SymbolicStokesData<2,3>>::ComputeGaussPointLHSContribution(
+    SymbolicStokesData<2,3> &rData,
+    MatrixType &rLHS)
+{
+    const double rho = rData.Density;
+    const double mu = rData.EffectiveViscosity;
+
+    const double h = rData.ElementSize;
+
+    const double dt = rData.DeltaTime;
+    const double bdf0 = rData.bdf0;
+
+    const double dyn_tau = rData.DynamicTau;
+
+    // Get constitutive matrix
+    const Matrix &C = rData.C;
+
+    // Get shape function values
+    const auto &N = rData.N;
+    const auto &DN = rData.DN_DX;
+
+    // Stabilization parameters
+    constexpr double stab_c1 = 4.0;
+    constexpr double stab_c2 = 2.0;
+
+    auto &lhs = rData.lhs;
+
+    double dt_inv = 0.0;
+    if (dt > 1e-09)
+    {
+        dt_inv = 1.0/dt;
+    }
+    if (fabs(bdf0) < 1e-9)
+    {
+        dt_inv = 0.0;
+    }

```

Comment:
You are checking twice the same statement. Please use an if & else statement.

Reply:
Besides, I don't think this needs to be checked for each element.

Reply:
> You are checking twice the same statement. Please use an if & else statement.

They are meant to be two different checks. Sorry for the confusion, have added a comment now.


> Besides, I don't think this needs to be checked for each element.

Agree, but how to do it.

Reply:
NA